tools/ai_sidecar/server.py

```
@@ -97,6 +119,88 @@ async def _safe_send(websocket: Any, payload: dict[str, Any]) -> None:
+            logger.info("broadcast send failed peer=%s err=%s",
+                        getattr(p, "remote_address", None), r)
+
+
+async def _safe_send_raw(websocket: Any, payload: str) -> None:
+    try:
+        await websocket.send(payload)
+    except Exception:
+        logger.exception("broadcast websocket send failed")
+
+
+async def _handle_external_frame(websocket: Any, data: dict[str, Any]) -> None:
```

Ruff (0.15.10)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `websocket`

(ANN401)